source/geometries/BlackBox.cc

```diff
@@ -156,53 +162,59 @@ namespace nexus {
 
   // SAPPHIRE //////////////////////////////////////////////////
 
-  G4String sapphire_name = "SAPPHIRE";
+  if (sapphire_) {
 
-  G4Material* sapphire_mat = materials::Sapphire();
+    G4String sapphire_name = "SAPPHIRE";
 
-  sapphire_mat->SetMaterialPropertiesTable(opticalprops::Sapphire());
+    G4Material* sapphire_mat = materials::Sapphire();
 
-  G4double pedot_thickn  = 150. *nm;
-  G4double sapphire_thickn  = 6. *mm;
-  G4double window_thickn  = sapphire_thickn + pedot_thickn;
-  G4double sapphire_diam = 20 *cm;
+    sapphire_mat->SetMaterialPropertiesTable(opticalprops::Sapphire());
 
-  G4Tubs* sapphire_solid_vol =
-    new G4Tubs(sapphire_name, 0, sapphire_diam/2.,
-               window_thickn/2., 0, twopi);
+    G4double pedot_thickn  = 150. *nm;
+    G4double sapphire_thickn  = 6. *mm;
+    G4double sapphire_diam = 20 *cm;
+    G4double window_thickn  = sapphire_thickn;
 
-  G4LogicalVolume* sapphire_logic_vol =
-    new G4LogicalVolume(sapphire_solid_vol, sapphire_mat, sapphire_name);
+    if (pedot_coating_) {
+    G4double window_thickn  = sapphire_thickn + pedot_thickn;}
```

Comment:
You are declaring this variable twice, and the second time it lives only in the local scope of `if (pedot_coating_)`, therefore its value is not changed in the rest of the code. Please, fix it.

Reply:
Done!